### DISCUSSION TITLE
Allow marking files as sources JARs for use in Maven

### DISCUSSION THREAD

**Discussion by BasiqueEvangelist** [modrinth/.github#176] [Feature requests]
Right now, the only special cased artifact in maven is the main artifact, [which is aliased to the primary file of the version.](https://github.com/modrinth/labrinth/blob/master/src/routes/maven.rs#L229-L233) This makes it easy to use mods through the maven, even if the author didn't specifically add support. However, using other classifiers (for example, `sources`) [requires naming the version file the way Maven expects it to be named.](https://docs.modrinth.com/maven#appendix-using-classifiers-sources-javadoc) (I think the docs aren't even entirely correct here, as sources jars need to specifically be `artifactid-version-sources.jar`)

Not only is this rather confusing (as some threads in #website-support can attest), this is also rather brittle: if the project's slug changes, or the version is referenced [in a different way](https://github.com/modrinth/labrinth/issues/624), the maven will stop serving it under the correct path and sources won't be linked.

## Proposed solution
Modrinth has a file type system, which is currently used for providing required/optional resource packs together with data packs. Adding a 'Sources' file type that the maven aliases to the correct sources path (and possibly even adding support for autopublishing sources jars from Minotaur) would make it *much* easier to provide sources on the Maven.
A special javadoc file type would also be a nice addition, although I don't think it's as useful as sources for developing against mods.

**Comment by dphaldes** [comment]
This problem can be seen in the project file https://modrinth.com/mod/blood-magic/version/3.3.2-44.
The jar name is "bloodmagic-1.20.1-3.3.2-44-sources.jar" and because it is not in the same format as Maven expects it, the sources are not fetched
